### ISSUE TITLE
Report some modification for reproduction.

### ISSUE THREAD

**Issue by katsura-jp** [siddk/vqa-outliers#2]
When I ran this repository, I needed to modify some scripts, so I reported them.

1. In README, when I extracted features of VQA2, the input/output folder's name was not correct.
https://github.com/siddk/vqa-outliers/blame/main/README.md#L85
```
x python scripts/extract.py --dataset vqa2 --images data/VQA-Images --spatial data/VQA-Spatials
o python scripts/extract.py --dataset vqa2 --images data/VQA2-Images --spatial data/VQA2-Spatials
```

2. When I extracted features of GQA, GQA's image path was not correct. The directory structure of GQA's images is not `GQA-Images/image/xxxx.jpg`, but `GQA-Images/xxxx.jpg`. So, it is necessary to modify the following lines.

https://github.com/siddk/vqa-outliers/blob/main/scripts/extract.py#L32
```
-        self.images = [x for x in os.listdir(os.path.join(path, "images")) if ".jpg" in x]
+        self.images = [x for x in os.listdir(os.path.join(path)) if ".jpg" in x]
```

https://github.com/siddk/vqa-outliers/blob/main/scripts/extract.py#L35
```
-        i_path = os.path.join(self.path, "images", self.images[index])
+        i_path = os.path.join(self.path, self.images[index])
```

3. This repository requires `pytorch-lightning==1.3.8` (https://github.com/siddk/vqa-outliers/blob/main/environment/environment-gpu.yaml#L127), but some lines did not match the version.
    1. In `training_epoch_end` of each model, it is necessary to remove return. So, I modified the following:
    ```
    -        return {"progress_bar": pbar, "log": log}
    +        for k, v in log.items():
    +            self.log(k, v)
    +
    +        # return {"progress_bar": pbar, "log": log}
    ```
    2. In `cartograph.py`, an argument of `ModelCheckpoint`, `filepath`, is not exist. I modified the following:
    ```
    checkpoint_callback = ModelCheckpoint(
    -        filepath=os.path.join(args.save_dir, "runs", run_name, args.mode + "-{epoch:02d}-{val_loss:.3f}-{val_acc:.3f}"),
    +        dirpath=os.path.join(args.save_dir, "runs", run_name),
    +        filename= args.mode + "-{epoch:02d}-{val_loss:.3f}-{val_acc:.3f}",
             monitor="val_acc",
    ```

4. To create Dataset Map, when I ran `cartograph.py`, it was necessary to add one augment, `--sync`, to log metrics information.
    https://github.com/siddk/vqa-outliers/blame/main/README.md#L118

NOTE: I didn't use conda, but docker + pip. The versions of packages are same to `environment-gpu.yaml`. Therefore, the behavior may be different from conda.

I should have created separate issues, but I apologize for putting them together.

**Comment by jahidhasanlinix** [comment]
Is this bugs fixed yet? @katsura-jp would you like to share the repo you used to reproduce the work after performing those modification? Thanks

**Comment by siddk** [comment]
Hey @jahidhasanlinix - to be honest I totally forgot about this! Should've set a reminder.

If you or @katsura-jp would like to submit a PR with the above fixes that might be the fastest, otherwise will try to get to this by Monday (still on mini-vacation)!

**Comment by katsura-jp** [comment]
Hi, @siddk and @jahidhasanlinix 

I have fixed most of the bugs, so I'm going to submit a PR.

**Comment by siddk** [comment]
Resolved in #4
